libpam/pam_modutil_private.h

```diff
@@ -17,6 +17,9 @@
 #define PWD_ABSURD_PWD_LENGTH  0x40001
 #define PWD_LENGTH_SHIFT 4 /* 2^4 == 16 */
 
+extern int
+_pam_add_alloc(pam_handle_t *pamh, void *data);
+
 extern void
 pam_modutil_cleanup(pam_handle_t *pamh, void *data,
```

Comment:
The naming convention for this new interface is slightly confusing.
While all other pam_modutil functions have `pam_modutil_` prefix, this one has `_pam_` prefix instead.

Also, `_pam_add_alloc` was declared in two different header files, we don't normally do that.

Reply:
I want to use it not just for modules or modutil releated functionality, but outside of it as well, e.g. for `_pam_mkargv`. If there is a better way to define it so it's visible from modutil* and libpam itself, without exposing it to everyone else, please let me know.